src/clj/rems/ext/mondo.clj

```diff
@@ -85,15 +93,26 @@
 
 (def ^:private supported-mondo-release-tag "the version of Mondo we support so far" "v2021-10-01")
 
-(defn- load-codes
-  "Load and index Mondo codes."
-  []
-  (->> (slurp (io/resource "mondo.edn"))
-       edn/read-string
-       (mapv (fn [[id label]] {:id (str "MONDO:" id) :label label}))
-       (index-by :id)))
-
 (def ^:private code-by-id (atom nil))
+(def ^:private codes-dag (atom nil))
+
+(defn- load-codes!
+  "Load and index Mondo codes, and generate directed acyclic graph
+   for querying Mondo code hierarchy."
+  []
+  (let [codes (->> (slurp (io/resource "mondo.edn"))
+                   edn/read-string
+                   (mapv (fn [[id label parents]]
+                           {:id (str "MONDO:" id)
+                            :label label
+                            :parents (mapv (partial str "MONDO:") parents)})))]
```

Comment:
seems like an easy optimization :+1:

Reply:
refactored so that mondo codes are preserved without prefix in-memory and prefixed when used from outside. this requires some extra code to do the conversion